certbot-ci/src/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -310,6 +313,47 @@ def test_graceful_renew_it_is_time(context: IntegrationTestsContext) -> None:
     assert_hook_execution(context.hook_probe, 'deploy')
 
 
+def test_renew_when_ari_says_its_time(context: IntegrationTestsContext) -> None:
+    """Test graceful renew is done when it is due time."""
+    certname = context.get_domain('renew')
+    context.certbot(['-d', certname])
+
+    assert_cert_count_for_lineage(context.config_dir, certname, 1)
+
+    # Tell Pebble to make ARI look urgent
+    with open(join(context.config_dir, 'live/{0}/cert.pem').format(certname), "r") as c:
+        certificate_pem = c.read()
+
+    ari_response = json.dumps({
+        "suggestedWindow": {
+           "start": "2020-01-01T00:00:00Z",
+           "end": "2020-01-01T00:00:00Z"
+        }
+    })
+
+    set_renewal_info_body = json.dumps(
+    {
+        "certificate": certificate_pem,
+        "ariResponse": ari_response
+    })
+
+    # POST to Pebble
+    misc.suppress_x509_verification_warnings()
+    url = PEBBLE_MANAGEMENT_URL + '/set-renewal-info/'
+    print(f"sending to {url}: {set_renewal_info_body}")
+    resp = requests.post(url, verify=False, timeout=10, data=set_renewal_info_body)
+    if resp.status_code != 200:
+        print(f"setting renewal info: {resp.status_code} {resp.text}")
+
+    assert resp.status_code == 200
```

Comment:
I would suggest moving this all (`# Tell Pebble to make ARI look urgent`...`assert resp.status_code == 200`) into `certbot-ci/src/certbot_integration_tests/utils/misc.py`; then `suppress_x509_verification_warnings` can stay in internal, but more importantly it can be used in other cases, and makes it clear what this test is actually testing